File: src/main/java/jb/files/FileReaderSameDataset.java
package jb.files;

import de.bwaldvogel.liblinear.InvalidInputDataException;
import de.bwaldvogel.liblinear.Problem;
import jb.config.Opts;
import jb.data.Dataset;

import java.io.File;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

public class FileReaderSameDataset implements FileHelper {

    @Override
    public Dataset readFile(Opts opts) throws IOException, InvalidInputDataException {
        Problem problem = Problem.readFromFile(new File(opts.getFilename()), 1);
        List<Problem> trainingProblems = new ArrayList<>();
        for (int i = 0; i < opts.getNumberOfBaseClassifiers(); i++) {
            trainingProblems.add(problem);
        }
        List<Problem> validatingProblems = new ArrayList<>();
        for (int i = 0; i < opts.getNumberOfSpaceParts(); i++) {
            validatingProblems.add(problem);
        }
        double minX = problem.x[0][0].getValue();
        double maxX = minX;
        for (int i = 1; i < problem.x.length; i++) {
            minX = Math.min(minX, problem.x[i][0].getValue());
            maxX = Math.max(maxX, problem.x[i][0].getValue());
        }
        return new Dataset(trainingProblems, validatingProblems, problem, minX, maxX);
    }

}

File: src/main/java/jb/files/FileReaderSameDataset.kt
package jb.files

import de.bwaldvogel.liblinear.InvalidInputDataException
import de.bwaldvogel.liblinear.Problem
import jb.config.Opts
import jb.data.Dataset

import java.io.File
import java.io.IOException
import java.util.ArrayList

class FileReaderSameDataset : FileHelper {

    @Throws(IOException::class, InvalidInputDataException::class)
    override fun readFile(opts: Opts): Dataset {
        val problem = Problem.readFromFile(File(opts.filePath!!), 1.0)
        val trainingProblems = ArrayList<Problem>()
        for (i in 0 until opts.numberOfBaseClassifiers) {
            trainingProblems.add(problem)
        }
        val validatingProblems = ArrayList<Problem>()
        for (i in 0 until opts.numberOfSpaceParts) {
            validatingProblems.add(problem)
        }
        var minX = problem.x[0][0].value
        var maxX = minX
        for (i in 1 until problem.x.size) {
            minX = Math.min(minX, problem.x[i][0].value)
            maxX = Math.max(maxX, problem.x[i][0].value)
        }
        return Dataset(emptyList(), 0.0, 0.0)
    }

}
